src/App.tsx

```diff
@@ -1,34 +1,87 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 import { Loader } from './components/Loader';
+import { TodoModal } from './components/TodoModal';
+import { SelectedFilterState } from './types/SelectedFilterState';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<null | Todo>(null);
+  const [selectedFilter, setSelectedFilter] = useState(SelectedFilterState.All);
+  const [query, setQuery] = useState('');
+
+  const filteredTodos = useMemo(() => {
+    const lowerCaseQuery = query.toLowerCase();
+
+    return todos
+      .filter(todo => {
+        switch (selectedFilter) {
+          case SelectedFilterState.Active:
+            return !todo.completed;
+          case SelectedFilterState.Completed:
+            return todo.completed;
+          default:
+            return true;
+        }
+      })
+      .filter(todo => (
+        todo.title.toLowerCase().includes(lowerCaseQuery)
+      ));
```

Comment:
Better to move this logic to separate helper function(outside the component and use this function here)